x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -220,6 +222,14 @@ export function App({
       return;
     }
 
+    if (usageCollection) {
+      usageCollection.reportUiCounter(
+        saveProps.returnToOrigin ? 'dashboard' : 'visualize',
```

Comment:
Here you could actually use 
``` typescript
saveProps.returnToOrigin ? incomingState?.originatingApp : 'visualize',
```

I notice you've done something similar in Visualize. This is necessary because dashboard isn't the only app that the user could be returning to.

Reply:
Thank you, I didn't find `incomingState?.originatingApp`